zookeeper-server/src/main/java/org/apache/zookeeper/server/ZooKeeperServerMain.java

```diff
@@ -180,6 +180,10 @@ public void runFromConfig(ServerConfig config) throws IOException, AdminServerEx
 
             serverStarted();
 
+            if (config.registerShutdownHook()) {
+                Runtime.getRuntime().addShutdownHook(new Thread(this::shutdown));
+            }
```

Comment:
Could this interact with the main `shutdownLatch` (which we `await` for right below)?

That's triggered by running the `zkServer`'s `shutdown` https://github.com/apache/zookeeper/blob/master/zookeeper-server/src/main/java/org/apache/zookeeper/server/ZooKeeperServer.java#LL935C31-L935C31

Then we can use the common shutdown path in both cases.

Reply:
Good point. Don't need to add new threads, just countdown the shutdown latch and it should be fine. No need to touch `QuorumPeerMain` either.

Reply:
> No need to touch `QuorumPeerMain` either.

Ignore this one. That's the non-standalone case, so still needed.